sdlib/d/gc/tstate.d

```diff
@@ -76,6 +76,17 @@ public:
 		}
 	}
 
+	void detach() {
+		import d.gc.tcache;
+		assert(&threadCache.state is &this,
+		       "Cannot detach manually from outside the thread!");
+
+		// should only ever be called while in a normal thread state.
+		size_t s = RunningState;
+		auto succeeded = state.casWeak(s, SuspendState.Detached);
+		assert(succeeded);
```

Comment:
> First, you can't ensure cas won't fail, but second, you can't ensure the thread is in a running state as a signal can be sent at anytime.

No, this can't happen. If it got a signal, it would not be able to detach itself, it would be in the signal handler. Note from the assert above, only a thread can detach itself explicitly, it can't be done outside the thread itself.

Besides, there is an assert here, the point is to ensure you only call this while in the right state. I want the bomb to explode if you don't do it right.

I suppose I could call store, and then validate the state was running?

> This is also impossible to review in its form. The behavior of all detach in the codebase are now changed for this, but we don't know where they are.

I grepped, there is one spot. Any currently valid code would only call detach while in the `Signaled` state. and I changed that one spot already.

I'll try and fix the names so I don't have to change the existing one. I felt like `detach` was the correct name for this.